src/lib/stores/ens/ens.ts

```diff
@@ -0,0 +1,64 @@
+import type { AbstractProvider } from 'ethers';
+import { NETWORK_CONFIG } from '../wallet/network';
+import FailoverJsonRpcProvider from '$lib/utils/FailoverJsonRpcProvider';
+import filterFalsy from '$lib/utils/filter-falsy';
+
+async function isAddressContract(provider: AbstractProvider, address: string) {
+  return (await provider.getCode(address)) !== '0x';
+}
+
+export function getMainnetProvider() {
+  const mainnet = NETWORK_CONFIG[1];
+  return new FailoverJsonRpcProvider(
+    filterFalsy([mainnet.rpcUrl, mainnet.fallbackRpcUrl]),
+    undefined,
+    undefined,
+    { logger: console },
+  );
+}
+
+/**
+ * Safely reverse lookup an ENS name, ensuring that the resolved address is safe to use.
+ * Resolved addresses are considered unsafe if they are contracts on one chain but not the other.
+ * @param currentChainProvider A provider for the current chain
+ * @param mainnetProvider A provider connected to Ethereum Mainnet
+ * @param chainId The chain ID of the current network
+ * @param name The name to reverse-lookup
+ * @returns resolved address if exists and safe, otherwise undefined
+ */
+export async function safeReverseLookp(
+  currentChainProvider: AbstractProvider,
+  mainnetProvider: AbstractProvider,
+  chainId: number,
+  name: string,
+) {
+  if (chainId === 1) {
+    return await mainnetProvider.resolveName(name);
+  }
+
+  const address = await mainnetProvider.resolveName(name);
+  if (!address) return undefined;
```

Comment:
Just out of curiosity, why do we returned undefined in one case and the straight return value in another?

Reply:
You're right this doesn't make any sense. Fixing Monday